src/borg/archiver.py

```diff
@@ -1363,10 +1363,11 @@ def _list_repository(self, args, manifest, write):
         output_data = []
 
         for archive_info in manifest.archives.list_considering(args):
+            archive = Archive(repository, key, manifest, archive_info.name)
```

Comment:
i tried the first way, but it gets rather messy as ArchiveInfo is quite different from Archive.

trying call keys concept now...